sherpa/astro/tests/test_astro_diag_arfrmf.py

```diff
@@ -0,0 +1,1669 @@
+#
+#  Copyright (C) 2017  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+"""
+Test the instrument-handling code without requiring external
+data files.
+TODO:
+  - add grouping to the PHA
+  - ARF and RMF can have different grids; I believe this is supported
+  - no testing of the startup/teardown methods; should this be done here?
+  - no testing of the nested models (also not fully tested in the full suite?)
+  - no testing of pileup
+  - no testing of PSFModel (also not fully tested in the full suite?)
+"""
+
+import warnings
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+import pytest  # pytest >= 3.0 is needed
+
+from sherpa.models.model import ArithmeticModel, \
+    ArithmeticConstantModel, BinaryOpModel
+from sherpa.astro.instrument import ARF1D, ARFModelNoPHA, ARFModelPHA, \
+    Response1D, RMF1D, RMFModelNoPHA, RMFModelPHA, \
+    RSPModelNoPHA, RSPModelPHA
+from sherpa.fit import Fit
+from sherpa.astro.data import DataARF, DataPHA, DataRMF
+from sherpa.models.basic import Const1D, Polynom1D, PowLaw1D
+from sherpa.utils.err import DataErr
+from sherpa.utils.testing import requires_xspec
+from sherpa.utils import _utils, requires_data, requires_fits
```

Comment:
The `requires_data` and `requires_fits` sumbols need to be moved to the line above (ie got from `sherpa.utils.testing`).

Reply:
I don't want to commit the changes myself in case it conflicts with something @olaurino is doing (and I don't have time)